demo/src/app/docs/demo-section-components/demo-examples-section/examples.component.ts

```diff
@@ -17,7 +17,7 @@ export class ExamplesComponent {
 
   @HostListener('document:click', ['$event'])
   preventEmptyHrefNav(event: Event) {
-    if (event && event.target && (event.target as Element).getAttribute('href') === '#') {
+    if (event && event.target && (event.target as Element).getAttribute('href') === '#' || event.srcElement.parentElement && (event.srcElement.parentElement as Element).getAttribute('href') === '#') {
```

Comment:
what if we have following structure:
```
<a href="#">
   <p>Text <b>text</b></p>
</a>
```
when you click on `<b>` you'll be redirected to the homepage. 
you should look for all tree of parent elements, not just the closest parent